lib/properties.ts

```diff
@@ -175,37 +177,74 @@ export class CompilerProps {
      * @param {?function} fn - Transformation to give to each value found
      * @returns {*} Transformed value(s) found or fn(defaultValue)
      */
+
+    // A lot of overloads for a lot of different variants:
+    // const a = this.compilerProps(lang, property); // PropertyValue
+    // const b = this.compilerProps<number>(lang, property); // number
+    // const c = this.compilerProps(lang, property, 42); // number
+    // const d = this.compilerProps(lang, property, undefined, (x) => ["foobar"]); // string[]
+    // const e = this.compilerProps(lang, property, 42, (x) => ["foobar"]); // number | string[]
```

Comment:
Looking at the code, what's the code path for the `number` variant here? As far as I can see, if fn is defined, it will always return whatever fn returns, right? And then the difficulty would come in narrowing x's type. Am I missing something?

Reply:
When there's both a default value and function, and base is a single language, it should go through
```ts
    get<T extends PropertyValue, R>(
        base: string,
        property: string,
        defaultValue: Widen<T>,
        fn?: (item: PropertyValue, language?: any) => R,
    ): typeof defaultValue | R;
```

Reply:
Thank you for the catch, my apologies! (Deleted my previous comment because it was incorrect)

Reply:
No worries at all :)